#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "laser_filters"

gen = ParameterGenerator()

gen.add("tolerance", double_t, 0, "Interval for line",    0.2, 0.01,   1.0)

gen.add("max_distance", double_t, 0, "Range for platforms",    10.0, 1.0,   40.0)

gen.add("number_skipped_angle", int_t, 0, "Number of angle can be skipped",    1, 1,   20)

gen.add("threshold_coefficient", double_t, 0, "Threshold rate for line fitting",    1.0, 0.1,   5.0)

gen.add("angle_threshold", double_t, 0, "angle threshold for platforms' edge",    10.0, 0.1,   40.0)

exit(gen.generate(PACKAGE, "laser_filters", "PlatformFilter"))